cli/command/image/build.go

```diff
@@ -268,7 +308,59 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 		progressOutput = &lastProgressOutput{output: progressOutput}
 	}
 
-	var body io.Reader = progress.NewProgressReader(buildCtx, progressOutput, 0, "", "Sending build context to Docker daemon")
+	if dockerfileCtx != nil && buildCtx == nil {
+		buildCtx = dockerfileCtx
+	}
```

Comment:
At this point we've rewritten and swapped out these two variables I'm not sure what this branch is really handling.

This will be true `if ("dockerfile is from stdin" OR "trust is enabled") AND (...?)` . From what I can tell `buildCtx` is always not-nil at this point, no?